consensus/src/quorum_store/utils.rs

```diff
@@ -297,9 +299,28 @@ impl ProofQueue {
                         .observe((current_time.round() - expiration.round()) as f64);
                 }
             }
-            size = size - 1;
+            // go through the remaining proofs, record the number of unexpired proofs
+            if block_finished {
+                if *expiration >= current_time && !excluded_proofs.contains(digest) {
+                    match self
+                        .digest_proof
+                        .get(digest)
+                        .expect("Entry for unexpired digest must exist")
+                    {
+                        Some((_, local)) => {
+                            remaining_proof_size += 1;
+                            if *local {
+                                remaining_local_proof_size += 1;
+                            }
+                        },
+                        None => {}, // Proof was already committed, skip.
+                    }
+                }
+            }
+            counters::NUM_BATCH_LEFT_WHEN_PULL_FOR_BLOCK.observe(remaining_proof_size as f64);
+            counters::NUM_LOCAL_BATCH_LEFT_WHEN_PULL_FOR_BLOCK.observe(remaining_local_proof_size as f64);
```

Comment:
Can we add counters here for block size (cur_txns, cur_bytes and ret.len())?

It is proposer based counter